web_server/authors/urls.py

```diff
@@ -18,11 +18,11 @@
          name="friend_checking_and_retrieval_of_author_id"),
     path('<str:author1_id>/friends/<path:author2_id>/',
          views.check_if_two_authors_are_friends, name="check_if_two_authors_are_friends"),
-    path('<str:author_id>/', views.retrieve_author_profile,
+    path('<slug:author_id>/', views.retrieve_author_profile,
```

Comment:
I notice in some areas you are using slug, and in others we are using path. Is there a difference between the two?